api/envoy/config/filter/http/compressor/gzip/v2/gzip.proto

```diff
@@ -0,0 +1,62 @@
+syntax = "proto3";
+
+package envoy.config.filter.http.compressor.gzip.v2;
+
+import "google/protobuf/wrappers.proto";
+
+import "udpa/annotations/migrate.proto";
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.config.filter.http.compressor.gzip.v2";
+option java_outer_classname = "GzipProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_migrate).move_to_package =
+    "envoy.extensions.filters.http.compressor.gzip.v3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: Gzip]
+// [#extension: envoy.filters.http.compressor.gzip]
+
+// [#next-free-field: 10]
+message Gzip {
+  enum CompressionStrategy {
+    DEFAULT = 0;
+    FILTERED = 1;
+    HUFFMAN = 2;
+    RLE = 3;
+  }
+
+  message CompressionLevel {
+    enum Enum {
+      DEFAULT = 0;
+      BEST = 1;
+      SPEED = 2;
+    }
+  }
```

Comment:
Why is the enum wraped with a message?

Reply:
I presume this is because there is already `DEFAULT` value in `enum CompressionStrategy`. So the second `DEFAULT` was given its own namespace.
This format is copied from the current gzip filter.

Do you mind if I rename the items to reflect zlib's name directly? Like
```protobuf
message Gzip {
  enum CompressionStrategy {
    // corresponds to Z_DEFAULT_STRATEGY
    DEFAULT_STRATEGY = 0;
    // corresponds to Z_FILTERED
    FILTERED = 1; 
    // corresponds to Z_HUFFMAN_ONLY
    HUFFMAN_ONLY = 2;
    // corresponds to Z_RLE
    RLE = 3;
  }

  enum CompressionLevel {
    // corresponds to Z_DEFAULT_COMPRESSION
    DEFAULT_COMPRESSION = 0;
    // corresponds to Z_BEST_COMPRESSION 
    BEST_COMPRESSION = 1; 
    // corresponds to Z_BEST_SPEED
    BEST_SPEED = 2;
  }
...
}
```
Alternatively `compression_level` could be just an integer with valid values from 1 to 9 where 1 is the best compression (but slow), 9 is the best speed (but low compression) and the default is 6. Not as descriptive, but more flexible.

Reply:
The new structure you have looks great. Re: compression level, I think we should either have 1-9 integer or an enum that can grow to express this (ideally with enum values matching the corresponding gzip compression level numberings.

Reply:
`Envoy::Compressor::ZlibCompressorImpl::CompressionLevel` is defined as enum, so I left this field  enum too, but extended with `COMPRESSION_LEVEL_X` values.